base_layer/wallet/src/wallet.rs

```diff
@@ -85,29 +85,33 @@ pub struct WalletConfig {
     pub comms_config: CommsConfig,
     pub factories: CryptoFactories,
     pub transaction_service_config: Option<TransactionServiceConfig>,
+    pub output_manager_service_config: Option<OutputManagerServiceConfig>,
     pub buffer_size: usize,
     pub rate_limit: usize,
     pub network: Network,
     pub base_node_service_config: Option<BaseNodeServiceConfig>,
 }
 
 impl WalletConfig {
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
         comms_config: CommsConfig,
         factories: CryptoFactories,
         transaction_service_config: Option<TransactionServiceConfig>,
+        output_manager_service_config: Option<OutputManagerServiceConfig>,
         network: Network,
         base_node_service_config: Option<BaseNodeServiceConfig>,
+        buffer_size: Option<usize>,
+        rate_limit: Option<usize>,
     ) -> Self
     {
         Self {
             comms_config,
             factories,
             transaction_service_config,
-            // This is the default buffer size for the pubsub_connector for the mobile wallet
-            buffer_size: 100,
-            // This is the default rate limit fot the pubsub_connector for the mobile wallet
-            rate_limit: 5,
+            output_manager_service_config,
+            buffer_size: buffer_size.unwrap_or_else(|| 1500),
+            rate_limit: rate_limit.unwrap_or_else(|| 1000),
```

Comment:
I think we should be careful with these; it will impact the memory footprint in the mobile wallet. The rate limit of `1000` proposed here is not balanced with the buffer size; better to throttle it a bit and not have messages dropped under stress conditions. I would propose `50`, then the `1500` buffer will fill up in 30 s if nothing is processed.